soh/soh/Enhancements/Cheats/MoonJump.cpp

```diff
@@ -0,0 +1,23 @@
+#include <libultraship/bridge.h>
+#include "soh/Enhancements/game-interactor/GameInteractor_Hooks.h"
+#include "soh/ShipInit.hpp"
+
+extern "C" {
+#include "macros.h"
+extern PlayState* gPlayState;
+}
+
+#define CVAR_NAME "gCheats.MoonJumpOnL"
+#define CVAR CVarGetInteger(CVAR_NAME, 0)
```

Comment:
I'm not a fan of these defines, specially `CVAR`, as it abstracts the fact that the third argument of `COND_ID_HOOK` is an expression and doesn't necessarily have to be a cvar get.

Reply:
In my experience(we have about 240 instances of this so far on 2ship), this is all the expression is most of the time, with the exception being things that are forced on for rando and/or boss rush, in which it’s pretty clear with `CVAR || IS_RANDO || IS_BOSS_RUSH`

the benefits are only typing out the cvar string and cvar default once, both of which have been victims of developer error, along with making the entire file more “copy/paste” friendly, for people creating new enhancements

Reply:
I'd prefer if we:
- Removed the `CVAR` macro and just used `CVarGetInteger(CVAR_NAME, 0)` since it's clearer imo and doesn't affect copy/paste (for the cvar default, see at the bottom)
- Maybe renamed `CVAR_NAME` to a specific name (`CVAR_MOON_JUMP` in this case) even tho it'd make it less copy/paste friendly as I think it'd make it easier to read at a glance (not 100% set on this one)

In terms of the cvar's default that's something that we should address more generally but for this we could do a `CVAR_DEFAULT` (or `CVAR_MOON_JUMP_DEFAULT` depending on the second earlier point) until we do the cvar refactor.

Reply:
I'll throw in my 2 cents but honestly I don't feel very strongly about this discussion in the first place because this feels very much like a personal preference thing.

How about we get 3 defines to clarify intent:
CVAR_MOON_JUMP_NAME
CVAR_MOON_JUMP_DEFAULT
CVAR_MOON_JUMP_VALUE